packages/convex-helpers/README.md

```diff
@@ -630,3 +630,74 @@ export const lastCountLongerThanName = query({
   },
 });
 ```
+
+## Manual Pagination
+
+Note Convex provides built-in pagination through `.paginate()` and
+`usePaginatedQuery()`.
+
+The `getPage` helper gives you more control of the pagination. You can specify
+the index ranges or do multiple paginations in the same query.
+
+However, you have to handle edge cases yourself, as described in
+https://stack.convex.dev/fully-reactive-pagination.
+
+More details and patterns will appear in upcoming articles.
+
+### Examples
+
+Fetch the first page, by creation time:
+
+```js
+const { page, indexKeys, hasMore } = await getPage(ctx, {
+  table: "messages",
+});
+```
+
+Fetch the next page:
+
+```js
+const {
+  page: page2,
+  indexKeys: indexKeys2,
+  hasMore,
+} = await getPage(ctx, {
+  table: "messages",
+  startIndexKey: indexKeys[indexKeys.length - 1],
+});
+```
+
+You can change the page size and order by any index:
+
+```js
+import schema from "./schema";
+const { page, indexKeys, hasMore } = await getPage(ctx, {
+  table: "users",
+  index: "by_name",
+  schema,
+  targetMaxRows: 1000,
+});
+```
+
+Fetch of a page between two fixed places in the index, allowing you to display
+continuous pages even as documents change and the queries reactively update:
+
+```js
+const { page } = await getPage(ctx, {
+  table: "messages",
+  startIndexKey,
+  endIndexKey,
```

Comment:
Is this how you'd do it? I have to wrap my head a little around the array of index keys. I'm pretty sure the first one is the absolute lower bound, and the last one is the absolute upper bound, but only after learning how it works
```suggestion
  startIndexKey: indexKeys[0],
  endIndexKey: indexKeys2[0],
```

Reply:
When refetching you don't change `startIndexKey` -- it's the same as when you originally fetched. And you set `endIndexKey: indexKeys[indexKeys.length-1]` 

This works because the start is exclusive and the end is inclusive. This allows refetching without needing to fetch the subsequent page (you don't need `indexKeys2`)

Reply:
I'll leave it as is. I thought it was specifically targeting the "reconnect to avoid gaps" usecase but it's more general and I don't want to confuse things.